### PR TITLE
Add more beta notice items and adjust TZNG EOS text

### PR DESCRIPTION
**Pull request by philrz**
@alfred-landrum @mccanne: Per our discussion this morning, here I've made some adjustments to the ZNG spec to line up with loose ends we're holding off on implementing in zq for now:

* TZNG EOS
* ZQL syntax for `enum`
* `float16`, `float32`, and `decimal`

I also adjusted the TZNG EOS text to emphasize that it's recommended, but optional.

I've included references to open issues in case users stumble onto these bullet items and want to express their use cases for adding them sooner.

### CODE REVIEW

File: zng/docs/spec.md [hunk 1]
```diff
@@ -3,18 +3,19 @@
 > ### Note: This specification is in BETA development.
 > We hope that no backward incompatible changes will be made during
 > the BETA phase.  We plan to
-> declare the specification stable and finalized in spring 2021.
+> declare the specification stable and finalized in Spring, 2021.
```
Comment (nwt) [inline zng/docs/spec.md:6]: Neither capitalization nor comma is conventional here.

File: zng/docs/spec.md [hunk 2]
```diff
@@ -663,10 +664,11 @@ A TZNG end-of-stream marker has the following form:
 ```
 #eos
 ```
-A TZNG stream or file should always be terminated with end-of-stream.
-This clears all of the previous type tag bindings and aliases, allowing
-multiple TZNG files with overlapping tags to be concatenated without the
-tags colliding.
+
+It is recommended that a TZNG stream or file be terminated with an
```
Comment (nwt) [inline zng/docs/spec.md:668]: Nit: I think the active voice of "A TZNG stream or file should" reads better.

File: zng/docs/spec.md [hunk 3]
```diff
@@ -663,10 +664,11 @@ A TZNG end-of-stream marker has the following form:
...
+end-of-stream marker. This explicitly clears all of the previous type tag
+bindings and aliases, allowing multiple TZNG files with overlapping tags to be concatenated without the tags colliding. However, a TZNG reader should not
```
Comment (nwt) [inline zng/docs/spec.md:670]: Nit: Wrap this.
```suggestion
bindings and aliases, allowing multiple TZNG files with overlapping tags to be
concatenated without the tags colliding. However, a TZNG reader should not
```

**Review by nwt — APPROVED** [review]

File: zng/docs/spec.md (hunk 1 above)
Comment (philrz) [reply zng/docs/spec.md:6]: Ah, right you are. I mistakenly thought it was a similar caps/comma rule as when writing month/day/year such as "October 21, 2020". But I see now I was wrong about that and have fixed.

**Pull request merged**